### PR TITLE
Sync with MoveIt 1

### PR DESCRIPTION
**Pull request by tylerjw**
- Prefer std::make_shared over new operator (#2756)
- Improve readability of comment
- Servo: minor fixups (#2759)

### CODE REVIEW

**Comment by henningkayser** [comment]
@tylerjw I don't think a rebase is really what we need for this. Could you reapply the merge and checkout the changes from this branch for resolving the conflicts?

**Comment by tylerjw** [comment]
> @tylerjw I don't think a rebase is really what we need for this. Could you reapply the merge and checkout the changes from this branch for resolving the conflicts?

Good catch.  I fixed this and it should now work as expected.

File: moveit_core/robot_trajectory/test/test_robot_trajectory.cpp [hunk 1]
```diff
@@ -152,6 +154,42 @@ TEST_F(RobotTrajectoryTestFixture, ModifyFirstWaypointByValue)
   modifyFirstWaypointAndCheckTrajectory(trajectory);
 }
 
+TEST_F(RobotTrajectoryTestFixture, DoubleReverse)
+{
+  robot_trajectory::RobotTrajectoryPtr trajectory;
+  initTestTrajectory(trajectory);
+  moveit_msgs::msgs::RobotTrajectory initial_trajectory_msg;
```
Comment (Abishalini) [inline moveit_core/robot_trajectory/test/test_robot_trajectory.cpp:161]:
```suggestion
  moveit_msgs::msg::RobotTrajectory initial_trajectory_msg;
```

File: moveit_core/robot_trajectory/test/test_robot_trajectory.cpp [hunk 2]
```diff
@@ -152,6 +154,42 @@ TEST_F(RobotTrajectoryTestFixture, ModifyFirstWaypointByValue)
...
+  trajectory->getRobotTrajectoryMsg(initial_trajectory_msg);
+
+  trajectory->reverse().reverse();
+
+  moveit_msgs::msgs::RobotTrajectory edited_trajectory_msg;
```
Comment (Abishalini) [inline moveit_core/robot_trajectory/test/test_robot_trajectory.cpp:166]:
```suggestion
  moveit_msgs::msg::RobotTrajectory edited_trajectory_msg;
```

File: moveit_core/robot_trajectory/test/test_robot_trajectory.cpp [hunk 3]
```diff
@@ -152,6 +154,42 @@ TEST_F(RobotTrajectoryTestFixture, ModifyFirstWaypointByValue)
...
+  trajectory->getRobotTrajectoryMsg(edited_trajectory_msg);
+
+  EXPECT_EQ(initial_trajectory_msg, edited_trajectory_msg);
+}
+
+TEST_F(RobotTrajectoryTestFixture, ChainEdits)
+{
+  robot_trajectory::RobotTrajectoryPtr initial_trajectory;
+  initTestTrajectory(initial_trajectory);
+  moveit_msgs::msgs::RobotTrajectory initial_trajectory_msg;
```
Comment (Abishalini) [inline moveit_core/robot_trajectory/test/test_robot_trajectory.cpp:176]:
```suggestion
  moveit_msgs::msg::RobotTrajectory initial_trajectory_msg;
```

**Comment by henningkayser** [comment]
Thanks, I'll review this tomorrow. Do you mind changing the merge commit message to `Merge https://github.com/ros-planning/moveit/commit/216d8ff614e6e25587ddbc6829582d428772e4ec`? This produces a nice link directly to the synced commit in GitHub and we don't need the information about the temporary branch.

File: moveit_core/robot_trajectory/test/test_robot_trajectory.cpp (hunk 1 above)
Comment (tylerjw) [reply moveit_core/robot_trajectory/test/test_robot_trajectory.cpp]: fixed, thank you

**Comment by tylerjw** [comment]
> Merge https://github.com/ros-planning/moveit/commit/216d8ff614e6e25587ddbc6829582d428772e4ec

Done

**Comment by tylerjw** [comment]
> Thanks, I'll review this tomorrow. Do you mind changing the merge commit message to `Merge https://github.com/ros-planning/moveit/commit/216d8ff614e6e25587ddbc6829582d428772e4ec`? This produces a nice link directly to the synced commit in GitHub and we don't need the information about the temporary branch.

done

**Review by henningkayser — APPROVED** [review]
Perfect, the diffs are nearly identical to MoveIt 1. There were only some changes that have already been applied for MoveIt 2. I'm pretty sure that the CI failure is a caching issue, Rolling has just been synced. The main branch doesn't seem to be affected, but the PRs might fail the rolling build

**Pull request merged**